stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -715,11 +716,56 @@ public class ChatClient internal constructor(
         return api.addDevice(device)
     }
 
+    @Deprecated(
+        message = "Use the searchMessages method with unwrapped parameters instead",
+        replaceWith = ReplaceWith("searchMessages(channelFilter, messageFilter, offset, limit)"),
+        level = DeprecationLevel.WARNING,
+    )
     @CheckResult
     public fun searchMessages(request: SearchMessagesRequest): Call<List<Message>> {
         return api.searchMessages(request)
     }
 
+    /**
+     * Search messages across channels. There are two ways to paginate through search results:
+     *
+     * * Using [limit] and [offset] parameters
+     * * Using [limit] and [next] parameters
+     *
+     * Pages of sort results will be returned with **next** and **previous** strings which
+     * can be supplied as a next parameter when making a query to get a new page of results.
+     *
+     * @param channelFilter channel filter conditions
+     * @param messageFilter message filter conditions
+     * @param offset pagination offset, cannot be used with sort or next
+     * @param limit the number of messages to return
+     * @param next pagination parameter, cannot be used with non-zero offset
+     * @param sort sort parameters, cannot be used with non-zero offset
```

Comment:
This looks a little bit weird with the double `sort` word. Maybe we can add `-` after the parameter name? What do you think?

Reply:
We have an internal doc where Marton explains why are no longer using dashes in such cases. I paraphrased the sentence a bit.

Reply:
Yes, please don't add a dash there - it creates a weird looking list item as it's interpreted as Markdown.

![image](https://user-images.githubusercontent.com/12054216/132262435-4df406e7-24e5-460d-b7f2-c4d650d978d5.png)

Reply:
On another note, @bychkovdmitry, can we please have full sentences for each of these comments, to adhere to the aforementioned guidelines? E.g.

```
     * @param sort The sort criteria applied to the result, cannot be used with non-zero offset.
```

Reply:
Yes, just a sec. This PR was created before the guidelines :)